web/app/charts/MapEditor.js

```diff
@@ -0,0 +1,457 @@
+                if(key == 13) {
+                    panel.title.innerHTML = this.value;
+                    defaultConfig.chartTitle = this.value;
+                }
+            };
+            this.appendChild(input);
+            input.focus();
+        };
+        document.body.addEventListener('click', function(e) {
+            var input = panel.title.childNodes[0];
+            var className = e.target.className;
+            if(input.nodeType === 1 && className != "form-input") {
+                panel.title.innerHTML = input.value;
+                defaultConfig.chartTitle = input.value;
+            };
+        });
+
+        onRender(previewArea, function(){
+            initializeChartSpace();
+        });
+    };
+
+
+  //**************************************************************************
+  //** update
+  //**************************************************************************
+    this.update = function(mapConfig, inputs){
+        me.clear();
+        for (var i=0; i<inputs.length; i++){
+            var input = inputs[i];
+            if (input!=null) inputs[i] = d3.csvParse(input);
+        }
+        if(mapConfig !== null && mapConfig !== undefined){
+            Object.keys(mapConfig).forEach(val=>{
+                chartConfig[val] = mapConfig[val]? mapConfig[val]:null;
+            });
+            panel.title.innerHTML = mapConfig.chartTitle;
+            }
+        inputData = inputs;
+        createDropDown(optionsDiv);
+        createOptions();
+    };
+
+
+  //**************************************************************************
+  //** createDropDown
+  //**************************************************************************
+    var createDropDown = function(parent){
+
+        var table = createTable();
+        var tbody = table.firstChild;
+        table.style.height = "";
+        parent.appendChild(table);
+        createMapDropDown(tbody);
+    };
+
+  //**************************************************************************
+  //** createMapDropDown
+  //**************************************************************************
+    var createMapDropDown = function(tbody){
+        dropdownItem(tbody,"mapType","Map Type",showHideDropDowns,mapInputs,"mapType");
+        dropdownItem(tbody,"latitude","Latitude",createMapPreview,mapInputs,"lat");
+        dropdownItem(tbody,"longitude","Longitude",createMapPreview,mapInputs,"long");
+        dropdownItem(tbody,"mapValue","Value",createMapPreview,mapInputs,"mapValue");
+
+        var tr, td;
+
+        tr = document.createElement("tr");
+        tr.id="projection";
+        tbody.appendChild(tr);
+        td = document.createElement("td");
+        tr.appendChild(td);
+        td.innerHTML= "Projection:";
+
+        tr = document.createElement("tr");
+        tbody.appendChild(tr);
+        td = document.createElement("td");
+        tr.appendChild(td);
+
+        mapProjection = new javaxt.dhtml.ComboBox(td, {
```

Comment:
Why is the mapProjection pulldown instantiated differently than the other pulldowns?

Reply:
Its instantiated differently because it assigns two values to the chartConfig instead of just one.